src/coreclr/jit/lower.cpp

```diff
@@ -4308,6 +4308,7 @@ GenTree* Lowering::OptimizeConstCompare(GenTree* cmp)
             // Transform EQ|NE(AND(x, y), y) into EQ|NE(AND(NOT(x), y), 0) when y is a constant.
             //
 
+            andOp1->ClearContained();
```

Comment:
Not really related to this PR, but this feels like the type of transform we could/should be doing in HIR, such as part of morph. This isn't something that is architecture or lowering specific and would likely mitigate some risk from the LIR/containment complexities.

It's also a transform that we could optimize more for constants that are powers of 2. In particular we can use `shr eax, n; and eax, 1` if setting the flag in a register and `bt eax, n` if consuming it as a flag.